code/modules/mob/dead/observer/observer.dm

```diff
@@ -418,12 +418,20 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		// Update med huds
 		current_mob.med_hud_set_status()
 		current_mob.log_message("had their player ([key_name(src)]) do-not-resuscitate / DNR", LOG_GAME, color = COLOR_GREEN, log_globally = FALSE)
+		//SKYRAT EDIT ADDITION - DNR TRAIT (Technically this is just to fix ghost-DNR'ing not actually DNR'ing, but it pairs with the trait so)
+		if(!current_mob.has_quirk(/datum/quirk/dnr))
+			current_mob.add_quirk(/datum/quirk/dnr)
+		//SKYRAT EDIT ADDITION END - DNR TRAIT
 	log_message("has opted to do-not-resuscitate / DNR from their body ([current_mob])", LOG_GAME, color = COLOR_GREEN)
 
 	// Disassociates observer mind from the body mind
 	mind = null
 
 	to_chat(src, span_boldnotice("You can no longer be brought back into your body."))
+	//SKYRAT EDIT ADDITION - DNR TRAIT - If they really somehow need back in, might as well have the message of how an admin can solve it
+	to_chat(src, span_italics("If this was a mistake in any way, ask an admin to either 'proc_call revive(TRUE, TRUE)' to fullheal you and keep the DNR trait, or 'vv, remove the DNR Quirk and do_not_revive Trait, call med_hud_set_status(), then have someone do defib/revival surgery'."))
+	//SKYRAT EDIT ADDITION END - DNR TRAIT
```

Comment:
I feel like this is unnecessary, if people make the mistake they're gonna ask admins to help either way

Reply:
Yea, I figured that'd be the consensus, it's more there so that admins know how than for the player; wish that calling remove_quirk properly removed 'em but alas
I'll just leave it out and have it in fat letters in the PR body, if admins need to know then someone'll come looking for the PR i hope